src/ZoneServer/Buffs/Handlers/EnchantEarth_Buff.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System.Threading.Tasks;
+using System;
+using Melia.Shared.Game.Const;
+using Melia.Zone.Buffs.Base;
+using Melia.Zone.Scripting;
+using Melia.Zone.World.Actors.Characters;
+using Melia.Zone.World.Actors.Characters.Components;
+
+namespace Melia.Zone.Buffs.Handlers
+{
+	/// <summary>
+	/// Handle for the Enchant Earth Buff, which increases the target's block penetration
+	/// </summary>
+	[BuffHandler(BuffId.EnchantEarth_Buff)]
+	public class EnchantEarth_Buff : BuffHandler
+	{
+		private const string VarName = "Melia.BlockPenetrationBonus";
+		private const string DifferentMapVarName = "Melia.EnchantEarth_Buff.DifferentMap";
+
+		public override void WhileActive(Buff buff)
+		{
+			// End the Buff if the Caster is Disconnected
+			if (buff.Caster is Character casterCharacter  && !casterCharacter.Connection.LoggedIn)
+			{
+				buff.End();
+			}
+
+			// End the Buff if the Caster is Dead
+			if (buff.Caster.IsDead)
+			{
+				buff.End();
+			}
+
+			// Ends the Buff if the Caster and Target are not in the same map
+			if (buff.Caster.Map.Id != buff.Target.Map.Id)
+			{
+				var differentMapHelper = buff.Vars.GetInt(DifferentMapVarName);
```

Comment:
"And the problem with the counting tick approach is that you have no control over the buff tick time, which a user or even an update might change at any time, which would could trigger this code much earlier or much later than expected."
This is true, but it's how the game actually works - it specifically notes in the tooltip that the buff expires 21-30 seconds after separating from the caster (the buff's update time is 10 seconds)

Reply:
Right, but the default value of updateTime is 10, so it is equivalent unless tinkered with, and the updateTime for this buff has no use except for this calculation.  I feel like the only reason you'd change it is if you wanted a longer grace period, in which case maybe you'd set it to 30 or something for a 60-90 second window.  In this case, if we instead hardcoded the time, you'd have to make a code change to get this behavior.

Reply:
What matters is not what we find likely or sensible, but reality and future changes.

Fact is, if you change the update time, regardless of the reason, the description will no longer match the behavior, and we can't easily fix that from the server. Meaning the best solution for us to keep server and description in sync, as developers of the server alone, is arguably to specifically check for a duration.

Additionally, this was also meant as general advice, because I assume Felip mostly followed your instructions, potentially without considering these consequences or side effects, which could result in him or others doing this again on the next buff, where the effects could be much more critical from the get go, without users doing unsensible modifications.